src/main/groovy/com/lesfurets/jenkins/unit/global/lib/LocalSource.groovy

```diff
@@ -11,7 +11,14 @@ class LocalSource implements SourceRetriever {
 
     @Override
     List<URL> retrieve(String repository, String branch, String targetPath) {
```

Comment:
The real problem here is that the (abstract) concept of _SourceRetriever_ is too tightly coupled with the one of a (git) repository and that we are bending the implementation accordingly. Rather than always requiring a branch and ignoring it when it does not make sense (i.e. in line 18), it would be much better if the `retrieve` API had a more generic signature. For example, it could take take a single map instead of the `repository/branch` tuple. This would allow us to specialize the map according to the needs of the specific retriever.